cassandra/connection.py

```diff
@@ -852,6 +858,28 @@ def factory(cls, endpoint, timeout, *args, **kwargs):
         else:
             return conn
 
+    def _build_ssl_context_from_options(self, opts):
+        # Python >= 3.10 requires either PROTOCOL_TLS_CLIENT or PROTOCOL_TLS_SERVER so we'll get ahead of things by always
+        # being explicit
+        ssl_version = opts.pop('ssl_version', ssl.PROTOCOL_TLS_CLIENT)
+        cert_reqs = opts.pop('cert_reqs', ssl.CERT_REQUIRED)
+        rv = ssl.SSLContext(protocol=ssl_version)
+        rv.check_hostname = bool(opts.pop('check_hostname', False))
+        rv.options = cert_reqs
+
+        certfile = opts.pop('certfile', None)
+        keyfile = opts.pop('keyfile', None)
+        if certfile:
+            rv.load_cert_chain(certfile, keyfile)
+        ca_certs = opts.pop('ca_certs', None)
+        if ca_certs:
+            rv.load_verify_locations(ca_certs)
+        ciphers = opts.pop('ciphers', None)
+        if ciphers:
+            rv.set_ciphers(ciphers)
+
+        return rv
```

Comment:
Follows the idea roughly sketched out in [Python docs for ssl.wrap_socket()](https://docs.python.org/3.11/library/ssl.html#ssl.wrap_socket)